agreement-service.js

```diff
@@ -34,21 +46,29 @@ function factory() {
 
     service.groups[group].displayOrder.push(agreement);
     service.groups[group].displayOrder.sort(function(a, b) {
-      return service.groups[group][a].title.localeCompare(
-        service.groups[group][b].title);
+      return service.groups[group].agreements[a].title.localeCompare(
+        service.groups[group].agreements[b].title);
     });
   };
 
   service.registerGroup = function(group) {
     if(group) {
       service.groups[group] = {
-        header: 'Service Agreement'
+        header: 'Service Agreement',
+        agreements: {}
       };
       service.groups[group]._displayOrder =
         service.groups[group].displayOrder = [];
     }
   };
 
+  service.getAgreementIds = function(group) {
```

Comment:
Maybe another name here is more appropriate.  It returns a list of full URLs for all the agreements in a group.